src/subdomains/supporting/payment/services/transaction-helper.ts

```diff
@@ -193,16 +194,18 @@ export class TransactionHelper implements OnModuleInit {
   private async getTargetEstimation(
     inputAmount: number | undefined,
     outputAmount: number | undefined,
-    fee: number,
+    fee: FeeDto,
     minFee: number,
     from: Asset | Fiat,
     to: Asset | Fiat,
   ): Promise<TargetEstimation> {
     const price = await this.priceProviderService.getPrice(from, to);
 
     const percentFeeAmount =
-      outputAmount != null ? price.invert().convert((outputAmount * fee) / (1 - fee)) : inputAmount * fee;
-    const feeAmount = Math.max(percentFeeAmount, minFee);
+      outputAmount != null
+        ? price.invert().convert((outputAmount * fee.rate) / (1 - fee.rate))
+        : inputAmount * fee.rate;
```

Comment:
We define `totalFee = inputAmount * fee.rate + fee.fixed` and `outputAmount = inputAmount - totalFee`.

If the user specifies an output amount, we need to take this into account.